### ISSUE TITLE
Append symbol in the legend if the layer is invalid

### ISSUE THREAD

**Issue by Gustry** [3liz/lizmap-web-client#4397]
It would be nice if the legend could display either a `⚠` icon next to it (with a tooltip, different for the admin) if the `GetLegendGraphic` contains a `valid=False`.

It would allow to see quickly that this legend relies on broken layer : 
![image](https://github.com/3liz/lizmap-web-client/assets/1609292/24548301-d983-4db0-9c78-91fcba21eb72)

http://lizmap.local:8130/index.php/lizmap/service

```
repository=testsrepository
project=feature_toolbar
SERVICE=WMS
REQUEST=GetLegendGraphic
VERSION=1.3.0
FORMAT=application/json
LAYER=tramway_lines,tramway_stops,parent_layer,parent_layer_without_attribute_table,tramway_stop_with_parenthesis_and_spaces
STYLES=défaut,défaut,défaut,défaut,défaut
```

in the JSON output, the plugin is now providing a `valid=False` value : 

![image](https://github.com/3liz/lizmap-web-client/assets/1609292/738a6daa-62b5-4945-80d3-aeceb10cc1db)

The icon provided by Lizmap server is a warning icon. I tried to override as well the layer name by appending `⚠` at the end of the string.

But both string AND mainly the icon is overriden by LWC, I gues because of 
https://github.com/3liz/lizmap-web-client/blob/f766786df82c2911bb9059f8c54488655d151d52/assets/src/modules/state/Symbology.js#L58

CC @nboisteault @rldhont

**Comment by Gustry** [comment]
![image](https://github.com/3liz/lizmap-web-client/assets/1609292/2a3cdff5-86ba-4f15-879a-5d0506aba97c)

**Comment by guenterw** [comment]
I find this extension very helpful, I also wanted to suggest it. 
Is there a workaround for realising this right now?

**Comment by Gustry** [comment]
>  Is there a workaround for realising this right now?

This was a real screenshot from the new version of the plugin, you can do the upgrade ;-) 2.9.1 just released.

The only drawback, if the user has set this icon for production, to show warnings on his own map :) (traffic jam...)

**Comment by guenterw** [comment]
Thanks for the hint but unfortunately it doesn't work for me yet. 
In all 3 current versions. Do I have to set anything special? I am logged in as an admin. 
These two raster data files do not exist on the test server. However, the layers are displayed normally in the legend.

![image](https://github.com/3liz/lizmap-web-client/assets/8373176/49772dad-1cfd-44cc-8edd-dd423861f920)

From my point of view, it would be ideal if the admin would see the “!” and the normal user would not see the layer at all.

Project "Test Jagdkataster..."
https://cx21-6.webgis.biz/
https://cx21-7.webgis.biz/
https://cx21-8.webgis.biz/

**Comment by Gustry** [comment]
> From my point of view, it would be ideal if the admin would see the “!” and the normal user would not see the layer at all.

It's a work in progress on this.
We have some discussions.
Especially about the use of : 

* https://docs.qgis.org/3.34/en/docs/server_manual/config.html#environment-variables `QGIS_SERVER_IGNORE_BAD_LAYERS`
* https://docs.3liz.org/py-qgis-server/configuration.html#cache-strict-check `CACHE_STRICT_CHECK`

Indeed, for an admin, some tooltip must be added, all tools using an invalid layer must be checked : edition tool, attritube table, dataviz, form filter ... (all layers which can be configured in Lizmap plugin, or in QGIS, for Value relation, Relation reference, Atlas ...)

Can you share a screenshot of QGIS about these layers ?

Hum; I just tried a project I knew with a broken layer : 
![image](https://github.com/3liz/lizmap-web-client/assets/1609292/16aa82fe-6df5-41ac-a99b-48b3a232ae4a)

Indeed, the symbol is not at the correct level.

**Comment by Gustry** [comment]
In my legend, I can see the flag `valid=False`, but I don't see it in yours @guenterw : 

![image](https://github.com/3liz/lizmap-web-client/assets/1609292/c512d5f3-d96b-47b9-967f-78d048ef0907)

**Comment by guenterw** [comment]
I have just restarted the server, no change:
![image](https://github.com/3liz/lizmap-web-client/assets/8373176/14bfbbaa-e95e-477f-b9b8-07797c210ee4)


the aerial image "Reute 25.09.2023 20cm" is a simple raster file with the relative path "../luftbilder/2023_....tif in the original project. This file does not exist here on the test server cx21.
![image](https://github.com/3liz/lizmap-web-client/assets/8373176/90fa5dee-a76f-4616-9693-039dc1b65c3a)

The WebGIS layer "FNP incl. 1. + 4. Änd." is actually more complicated. It is a grouping of three raster files.
![image](https://github.com/3liz/lizmap-web-client/assets/8373176/72fe7bdc-308c-4312-91e8-defc92add5dd)

**Comment by Gustry** [comment]
> The WebGIS layer "FNP incl. 1. + 4. Änd." is actually more complicated. It is a grouping of three raster files.
This is not supported. I'm not sure how the legend is built on the QGIS Server side. That could be a dev or an idea.

But for sure, by reading the logs from your server, you will have a message like : 

> Skipping the layer 'FNP incl. 1. + 4. Änd.' because it's not a vector layer

> I have just restarted the server, no change:

As usual, you should read your QGIS Server logs. If needed, increase log level.
You will see the incoming HTTP request `GetLegendGraphic` (we can see it your in webbrowser like this 

![image](https://github.com/3liz/lizmap-web-client/assets/1609292/7a77f666-2601-4af8-93e5-665a0e5d9892)

I just made a commit to have a little bit more debug if an invalid layer has been found. If you can download the latest zip.
Either the ZIP directly, or you can use our custom QGIS repository with QGIS-Plugin-Manager
https://packages.3liz.org/

Thanks for your help

**Comment by Gustry** [comment]
The latest ZIP might be available in a dozen of minutes, just in case. We have some CI running to build the ZIP package.

**Comment by guenterw** [comment]
ok, Lizmap Server Plugin 2.9.2-alpha is active:
![image](https://github.com/3liz/lizmap-web-client/assets/8373176/7a63e772-d252-438c-bd25-2cf100d8979a)

If I understand it correctly, the "!" in the legend cannot work with the grouped layers (FNP...).
But it should work for the aerial view (Luftbild...)? 

Otherwise, I'm afraid I don't understand everything about logging. Unfortunately, I still have some projects to work on. Many thanks for the help/feedback.

**Comment by Gustry** [comment]
> If I understand it correctly, the "!" in the legend cannot work with the grouped layers (FNP...).

Yes, it would need some dev. I'm not sure about the current behavior.

> But it should work for the aerial view (Luftbild...)?

No, this was not yet supported. It's in progress in https://github.com/3liz/qgis-lizmap-server-plugin/pull/77

> Otherwise, I'm afraid I don't understand everything about logging

This is a must ;-)

That's the only way to see how your server is behaving, especially when it doesn't work as expected. QGIS Server core and plugins are verbose, about what they are doing etc.

**Comment by guenterw** [comment]
Thank you for the clarification.
It would of course be nice if it also worked for the raster data. But I should actually have my own vector and raster data under control.
Such a display for external WMS or WFS services would be more important to me. But that is probably even more complex?

With logging, I was referring more to the analysis in the browser. I am essentially familiar with logging from the QGIS server, even if I don't use it enough. But I also see myself more as a user and tester.

**Comment by guenterw** [comment]
I don't really know why, but now the “!” for raster files works. Maybe after the last update of the server plugin (2.9.2-alpha - 5098057) I had only restarted Apache and not the QGIS server.

![image](https://github.com/3liz/lizmap-web-client/assets/8373176/a67a9219-5cc1-4938-8cbb-185cd8f77475)

This is visible to everyone, which may be a good thing.
Thank you very much.

**Comment by Gustry** [comment]
Good to know. Still no luck for vector layer ? Can you just add typo in the datasource of "Jagdkataster" and/or "Politisch" ?

> This is visible to everyone, which may be a good thing.

It's a work in progress. The webbrowser is now aware about this invalid layer. And it's the purpose of this ticket : 

* for normal user : make the user that something, the tool is broken and Lizmap can't work properly (edition, attribute table, related table in form...)
* for admin : explaining the reason and some clues what to check etc

In your qgis logs, you should have the warning about the raster layer.
